### DISCUSSION TITLE
IDE Plugin that offers navigation between vector components

### DISCUSSION THREAD

**Discussion by nzxwang** [vectordotdev/vector#21371] [General]
I couldn't find any prior discussion around implementing an IDE plugin that would allow for easy navigation between vector components similar to how IntelliJ supports navigation between classes, functions, etc. Something like this would be very useful for a large and non-linear vector pipeline.

**Comment by pront** [comment]
Hi @nzxwang, I created an [issue](https://github.com/vectordotdev/vector/issues/21475) from this discussion. Feel free to add details there that might be helpful to future contributors.
